packages/pigeon/tool/shared/native_project_runners.dart

```diff
@@ -38,11 +44,23 @@ Future<int> runFlutterBuild(
 }
 
 Future<int> runXcodeBuild(
-  String nativeProjectDirectory, {
+  String nativeProjectDirectory,
+  String platform, {
   String? sdk,
   String? destination,
+  String? configuration,
   List<String> extraArguments = const <String>[],
 }) {
+  File? disabledSandboxEntitlementFile;
+  if (extraArguments.contains('test') &&
+      platform.toLowerCase() == 'macos' &&
+      Platform.environment['LUCI_CI'] == 'True') {
```

Comment:
@tarrinneal Do you think we should check if we're in CI here or just always disable sandboxing when running `xcodebuild test` on macOS? If we do it only in CI, hypothetically if you're running it locally, you'll see the pop up dialog and need to accept it.

Reply:
I would rather we not special-case CI when we can avoid it; differences between how tests run locally vs CI can make it frustrating for people to debug CI failures.